server/src/test/java/io/crate/lucene/WithinQueryBuilderTest.java

```diff
@@ -24,18 +24,28 @@
 import static io.crate.testing.Asserts.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
+import org.apache.lucene.document.ShapeField;
+import org.apache.lucene.search.ConstantScoreQuery;
 import org.apache.lucene.search.Query;
 import org.apache.lucene.spatial.prefix.WithinPrefixTreeQuery;
 import org.junit.Test;
 
 public class WithinQueryBuilderTest extends LuceneQueryBuilderTest {
 
```

Comment:
Was going to say that we need an adjustment in the `WithinFunction.toQuery` but then realized that we do that optimization only for `within(point, geo_shape_literal)`, ie shape is provided by user and is not read from the index so no adjustment is needed. 

Maybe worth adding a code comment next to https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/expression/scalar/geo/WithinFunction.java#L183 that Lucene query construction is index type agnostic for geo shapes since it's provided as Literal